tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/PathUtil.cs

```diff
@@ -0,0 +1,266 @@
+﻿// <copyright file="PathUtil.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests;
+
+public class PathUtil
+{
+    /// <summary>
+    /// Create a relative path from one path to another. Paths will be resolved before calculating the difference.
+    /// Default path comparison for the active platform will be used (OrdinalIgnoreCase for Windows or Mac, Ordinal for Unix).
+    /// </summary>
+    /// <param name="relativeTo">The source path the output should be relative to. This path is always considered to be a directory.</param>
+    /// <param name="path">The destination path.</param>
+    /// <returns>The relative path or <paramref name="path"/> if the paths don't share the same root.</returns>
+    /// <exception cref="ArgumentNullException">Thrown if <paramref name="relativeTo"/> or <paramref name="path"/> is <c>null</c> or an empty string.</exception>
+    public static string GetRelativePath(string relativeTo, string path)
+#if NETCOREAPP
+        => Path.GetRelativePath(relativeTo, path);
+#else
+    {
+        const char directorySeparatorChar = '\\';
+        const char altDirectorySeparatorChar = '/';
+        const char volumeSeparatorChar = ':';
+
+        const string extendedDevicePathPrefix = @"\\?\";
+        const string uncExtendedPathPrefix = @"\\?\UNC\";
+
+        // based on https://github.com/dotnet/runtime/blob/1d1bf92fcf43aa6981804dc53c5174445069c9e4/src/libraries/System.Private.CoreLib/src/System/IO/Path.cs#L843
+        var comparisonType = StringComparison.OrdinalIgnoreCase; // Windows only
+        relativeTo = Path.GetFullPath(relativeTo);
+        path = Path.GetFullPath(path);
+
+        // Need to check if the roots are different- if they are we need to return the "to" path.
+        if (!AreRootsEqual(relativeTo, path, comparisonType))
```

Comment:
isn't this redundant with the check below on common length ?

Reply:
Honestly, I didn't look to hard, I just copied from the .NET Core implementation 😄

Reply:
Ah ok, I thought it was a lot of work just for a test ^^
Maybe worth a comment at the top to say where it's copy/pasted from ?

Reply:
```diff
+        // based on https://github.com/dotnet/runtime/blob/1d1bf92fcf43aa6981804dc53c5174445069c9e4/src/libraries/System.Private.CoreLib/src/System/IO/Path.cs#L843
```

😃

Reply:
oh yeah right, it's just that function, not the whole file.